lib/whitehall/asset_manager_storage.rb

```diff
@@ -38,11 +38,14 @@ def store!(file)
 
   def retrieve!(identifier)
     asset_path = uploader.store_path(identifier)
-    File.new(asset_path)
+    File.new(asset_path, uploader.model, uploader.version_name)
   end
 
   class File
-    def initialize(asset_path)
+    def initialize(asset_path, model = nil, version_name = nil)
```

Comment:
Since the initializer has changed, shouldn't the `store!` method also return a `File.new(uploader.store_path, uploader.model, uploader.version_name)`?